crates/storage/src/alloc/boxed/storage.rs

```diff
@@ -29,21 +29,27 @@ use ink_primitives::Key;
 
 #[cfg(feature = "std")]
 const _: () = {
-    use crate::traits::StorageLayout;
-    use ink_metadata::layout::{
-        CellLayout,
-        Layout,
-        LayoutKey,
-    };
-
-    impl<T> StorageLayout for StorageBox<T>
+    impl<T> scale_info::TypeInfo for StorageBox<T>
     where
         T: SpreadLayout,
     {
-        fn layout(key_ptr: &mut KeyPtr) -> Layout {
-            Layout::Cell(CellLayout::new::<DynamicAllocation>(LayoutKey::from(
-                key_ptr.advance_by(1),
-            )))
+        fn type_info() -> scale_info::Type {
+            scale_info::Type::builder()
+                .path(
+                    scale_info::Path::from_segments(vec!["ink_storage", "alloc", "Box"])
+                        .expect("encountered invalid Rust path"),
+                )
+                // Unfortunately we cannot encode the type parameters of the box since they
+                // have to be `T: scale::Codec`. However, them not requiring to be encodable
+                // is the purpose of the storage `Box<T>`.
+                // Until we found a solution to this problem we cannot uncomment the below
+                // line of code:
+                //
+                // .type_params(vec![scale_info::MetaType::new::<T>()])
```

Comment:
What's the trade-off of not being able to un-comment the line?

Reply:
It does not make sense for the `T` in `ink_storage::alloc::Box<T>` to implement `scale_info::TypeInfo` so having this requirement on the bounds would not make sense, too. The `T: scale_info::TypeInfo` bound means that `T` is somehow `scale` encodable. However, `ink_storage::alloc::Box<T>` actually exists exactly for the purpose of allowing non-`scale` encodable instances within others.